client/components/relatedProducts/RelatedProducts.jsx

```diff
@@ -5,18 +5,21 @@ import axios from 'axios';
 import getImageUrl from '../../utils/getImageUrl';
 
 import Card from './Card';
+import ComparisonModal from './ComparisonModal';
 
 import '../../css/relatedProducts/RelatedProducts.scss';
 
-const RelatedProducts = ({ productId, rating }) => {
+const RelatedProducts = ({ currentProduct, rating }) => {
   const [products, setProducts] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
   const [isError, setIsError] = useState({ error: false, msg: '' });
   const [imageUrl, setImageUrl] = useState('');
+  const [showComparison, setShowComparison] = useState(false);
+  const [relatedProduct, setRelatedProduct] = useState(null);
 
   useEffect(() => {
     // make a GET request to /api/products/:product_id/related
-    axios.get(`/api/products/${productId}/related`)
+    axios.get(`/api/products/${currentProduct.id}/related`)
       .then(({ data }) => data)
       .then((ids) => ids.map((id) => axios.get(`/api/products/${id}`)))
       .then((promises) => {
```

Comment:
what do you do with these promises?

Reply:
These promises are getting the list of ids from `/api/products/:product_id/related` and mapping over them to get the related products and displaying the information in the cards.